Source/OCMock/OCMockObject.m

```diff
@@ -318,6 +318,12 @@ - (id)forwardingTargetForSelector:(SEL)aSelector
     {
         OCMRecorder *recorder = [[OCMMacroState globalState] recorder];
         [recorder setMockObject:self];
+        // If the initTarget has not been set then this is the first call through the
+        // recorder so we set it to the mock object.
+        if (![recorder initTarget])
```

Comment:
As far as I understand there is only one path to the situation where the recorder already has a target. This happens when a test uses `ClassMethod()` because that ends up calling `classMethod` on the recorder, via `switchToClassMethod` in `OCMMacroState`, and it does so before the invocation of the method. 

Looking at the [implementation of the macro](https://github.com/erikdoe/ocmock/blob/d79c405ceee1c97aba5b9977ad9000ac60749780/Source/OCMock/OCMock.h#L92-L96), it appears that the value of the expression is not the recorder but whatever the invocation returns.

Now, with how I understand this so far, wouldn't that mean we should actually overwrite `initTarget` here?

Reply:
You are correct that this block (321-326) can be reduced down to `[recorder setInitTarget:self];`. Technically it is incorrect in the `classMethod` case, but I'm trying to think of a case where we actually care about `classMethod` because `initTarget` is only going to be referred to in the case of an init family method which by definition can't be a class method.